test/integration/node-specific/resource_tracking_script_builder.ts

```diff
@@ -106,3 +132,70 @@ export async function runScript(
     heap
   };
 }
+
+/**
+ * A helper for running arbitrary MongoDB Driver scripting code in a resource information collecting script.
+ * This script uses info from node:process to collect resource information.
+ *
+ * **The provided function is run in an isolated Node.js process**
+ *
+ * A user of this function will likely need to familiarize themselves with the surrounding scripting, but briefly:
+ * - Many MongoClient operations (construction, connection, commands) can result in resources that keep the JS event loop running.
+ *   - Timers
+ *   - Active Sockets
+ *   - File Read Hangs
+ *
+ * - This function performs assertions that at the end of the provided function, the js event loop has been exhausted
+ *
+ * @param name - the name of the script, this defines the name of the file, it will be cleaned up if the function returns successfully
+ * @param config - `this.configuration` from your mocha config
+ * @param func - your javascript function, you can write it inline! this will stringify the function, use the references on the `this` context to get typechecking
+ * @param options - settings for the script
+ * @throws Error - if the process exits with failure or if the process' resources are not cleaned up by the provided function.
+ */
+export async function runScriptAndGetProcessInfo(
+  name: string,
+  config: TestConfiguration,
+  func: ProcessResourceTestFunction
+) {
+  const scriptName = `${name}.cjs`;
+  const scriptContent = await testScriptFactory(
+    name,
+    config.url(),
+    REPORT_RESOURCE_SCRIPT_PATH,
+    func
+  );
+  await writeFile(scriptName, scriptContent, { encoding: 'utf8' });
+  const logFile = name + '.logs.txt';
+
+  const script = spawn(process.argv[0], [scriptName], { stdio: ['ignore', 'ignore', 'ignore'] });
```

Comment:
non-blocking suggestion:

```suggestion
  const script = spawn(process.execPath, [scriptName], { stdio: ['ignore', 'ignore', 'ignore'] });
```

since that works even if something messes with the argument list, and maybe

```suggestion
  const script = spawn(process.execPath, [scriptName], { stdio: ['ignore', 'ignore', 'inherit'] });
```

since it might be a bit difficult to debug the inner process if something goes wrong otherwise?

Reply:
I committed the execPath suggestion! As for the stdio suggestion, we're redirecting all stderr output to a logs file and handling errors through there - if you see handling of `exitCode` in this file. What do you think?

Reply:
To be clear, I'm not necessarily suggesting that the JSON serialization goes away here, that might still be useful – but having stderr open and ending up _somewhere_ where it can be inspected is generally a good idea

Reply:
Making stderr inherit SGTM. It is different than the error logging we're doing to surface assertion errors or test code errors via the log file. Stderr will get printed to by node if something unexpected happens throwing an error outside of main, possibly in start up or in the then/catch handling

Reply:
That's a good point on error handling outside of main! Added it in